### PR TITLE
Update NDK to 26.0.10792818

### PR DESCRIPTION
**Pull request by snnn**
### Description
Update NDK to 26.0.10792818 which is included in every macOS build machine so that we do not need to download a different version every time in every build. 

### Motivation and Context
Downloading NDK on-the-fly is a main contributor of Android related build failures.

### CODE REVIEW

**Comment by edgchen1** [comment]
should we just update to the latest LTS version, r26?
https://developer.android.com/ndk/downloads#lts-downloads

**Comment by snnn** [comment]
It seems not easy.

**Review by mszhanyi — APPROVED** [review]

File: cmake/onnxruntime_unittests.cmake [hunk 1]
```diff
@@ -1553,7 +1553,7 @@ if (NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
     endif()
   endif()
 
-  if (onnxruntime_BUILD_SHARED_LIB AND (NOT onnxruntime_MINIMAL_BUILD OR onnxruntime_MINIMAL_BUILD_CUSTOM_OPS))
+  if (onnxruntime_BUILD_SHARED_LIB AND (NOT onnxruntime_MINIMAL_BUILD OR onnxruntime_MINIMAL_BUILD_CUSTOM_OPS) AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
```
Comment (edgchen1) [inline cmake/onnxruntime_unittests.cmake:1556]: `NOT ANDROID` is more concise
https://cmake.org/cmake/help/latest/variable/ANDROID.html

**Comment by skottmckay** [comment]
> It seems not easy.

What's not easy? The image includes 26.0.10792818, and the Android SDKs and NDKs appears to be the same versions across all the macos and linux images I looked at. 

https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

**Comment by skottmckay** [comment]
There are also predefined environment variables for the NDK path that we could maybe use instead of calculating our own NDK_PATH value

https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#environment-variables-2

**Comment by snnn** [comment]
New NDK version means new compiler/new linker version. As you can see there are some wired link errors after I made the change. Tomorrow I will try to reproduce them locally

**Comment by skottmckay** [comment]
Yeah it's odd. I have a local WSL2 setup with NDK 25.2 that I used a couple of months ago and it built ORT without a problem back then, and now it's complaining about this being missing (same as the CI):

https://github.com/microsoft/onnxruntime/blob/74a8acf4057902f15b574046b2c6b0e3a0b11e49/include/onnxruntime/core/framework/data_types.h#L250

It's defined in the framework library and that hasn't changed for years. Possibly some other cmake change is affecting the libraries the test exe is building against.

https://github.com/microsoft/onnxruntime/blob/74a8acf4057902f15b574046b2c6b0e3a0b11e49/onnxruntime/core/framework/data_types.cc#L1463-L1472) 

FWIW building with the 26.0 NDK works. https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1166226&view=logs&j=b27c03c3-79f9-5527-4b3a-dcfd5ff9627d&t=40e3fc77-3920-5dd0-c3e1-53eae25358ab

As the security guidance is to use the latest NDK maybe we should just try and do that instead of chasing down a link error that occurs with 25.2 but not 25.0 or 26.0 (or any other build of ORT).

**Comment by skottmckay** [comment]
> /onnxruntime_src/onnxruntime/test/onnx/TestCase.cc:189:7: error: variable 'item_id' set but not used [-Werror,-Wunused-but-set-variable]
  int item_id = 1;

May need to add `[[maybe_unused]]` to the item_id variable as the usage is in a catch block that gets thrown away in a build with no exceptions.

**Review by edgchen1 — APPROVED** [review]

**Review by jchen351 — APPROVED** [review]

**Pull request merged**